shared/rangeanalysis/codeql/rangeanalysis/ModulusAnalysis.qll

```diff
@@ -264,6 +264,13 @@ module ModulusAnalysis<
       val = remainder(val0 + delta, mod)
     )
     or
+    exists(Sem::Expr mid, int v, int m1, int m2 |
+      exprModulus(mid, b, v, m1) and
+      e = modExpr(mid, m2) and
+      mod = m1.gcd(m2) and
+      val = remainder(v, mod)
```

Comment:
Can you remind me of the math here? As far as I can tell, we have:
- `mid` = `b` + `v` (mod `m1`)
- `e` = mod(`mid`, `m2`)
- `mod` = gcd(`m1`, `m2`)
- `val` = mod (`v`, `mod`)

And we want to conclude that `e` = `b` + `c` (mod `mod`).

Reply:
Yes. Your second line implies `e` = `mid` (mod `m2`). Note that we generally have the following implication: If `x` = `y` (mod `k`) and `d` is a divisor of `k` then it also holds that `x` = `y` (mod `d`). Therefore, since `mod` divides both `m1` and `m2`, we have `e` = `mid` = `b` + `v` = `b` + `val` (mod `mod`).

Reply:
Btw. I see that I forgot to include `mod != 1` to rule out useless results.

Reply:
Thank you! That makes sense 🙇